### PR TITLE
[JENKINS-65398] Terminology update

### PR DESCRIPTION
**Pull request by twasyl**
Terminology update as part of [JENKINS-65398](https://issues.jenkins.io/browse/JENKINS-65398).

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/hudson/maven/AbstractMavenBuilder.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ protected AbstractMavenBuilder(BuildListener listener, Collection<MavenModule> m
     }
     
     /**
-     * Invoked after the maven has finished running, and in the master, not in the maven process.
+     * Invoked after the maven has finished running, and in the controlelr, not in the maven process.
```
Comment (Vlatombe) [inline src/main/java/hudson/maven/AbstractMavenBuilder.java:96]:
```suggestion
     * Invoked after the maven has finished running, and in the controller, not in the maven process.
```

File: src/main/java/hudson/maven/AbstractMavenProcessFactory.java [hunk 2]
```diff
@@ -549,9 +549,9 @@ public String call() throws IOException {
     }
 
     /**
-     * Copies a Maven-related JAR to the slave on demand.
-     * Can also be used when run on master.
-     * @param root the FS root of the slave (null means running on master)
+     * Copies a Maven-related JAR to the agent on demand.
+     * Can also be used when run on controller.
+     * @param root the FS root of the agent (null means running on controller)
```
Comment (Vlatombe) [inline src/main/java/hudson/maven/AbstractMavenProcessFactory.java:554]:
```suggestion
     * @param root the FS root of the agent (null means running on the built-in node)
```

File: src/main/java/hudson/maven/PlexusModuleContributorFactory.java [hunk 3]
```diff
@@ -20,9 +20,9 @@
  *
  * <h2>Lifecycle</h2>
  * <p>
- * {@link PlexusModuleContributorFactory}s are instantiated as singletons on the master, and whenever a new Maven
+ * {@link PlexusModuleContributorFactory}s are instantiated as singletons on the controller, and whenever a new Maven
  * process starts, its {@link #createFor(AbstractBuild)} method is called to instantiate {@link PlexusModuleContributor},
- * which gets  serialized on the master, then deserialized inside the Maven process, and then its
+ * which gets  serialized on the controller, then deserialized inside the Maven process, and then its
```
Comment (Vlatombe) [inline src/main/java/hudson/maven/PlexusModuleContributorFactory.java:25]:
```suggestion
 * which gets serialized on the controller, then deserialized inside the Maven process, and then its
```

File: src/main/java/hudson/maven/RedeployPublisher.java [hunk 4]
```diff
@@ -237,7 +237,7 @@ private MavenEmbedder createEmbedder(TaskListener listener, AbstractBuild<?,?> b
                 Node buildNode = build.getBuiltOn();
                 
                 if(buildNode == null) {
-                    // assume that build was made on master
+                    // assume that build was made on controller
```
Comment (Vlatombe) [inline src/main/java/hudson/maven/RedeployPublisher.java:240]:
```suggestion
                    // assume that build was made on the built-in node
```

File: src/main/java/hudson/maven/RedeployPublisher.java [hunk 5]
```diff
@@ -247,7 +247,7 @@ private MavenEmbedder createEmbedder(TaskListener listener, AbstractBuild<?,?> b
                     altSettingsPath = remoteUserHome + "/.m2/settings.xml";
                 }
                 
-                // we copy this file in the master in a  temporary file 
+                // we copy this file in the controller in a  temporary file
```
Comment (Vlatombe) [inline src/main/java/hudson/maven/RedeployPublisher.java:250]:
```suggestion
                // we copy this file in the controller in a temporary file 
```

File: src/main/resources/hudson/maven/reporters/Messages_de.properties [hunk 6]
```diff
@@ -27,27 +27,27 @@ HistoryWidgetImpl.Displayname=Entwicklungsverlauf
 MavenAbstractArtifactRecord.Displayname=Artefakte ausbringen (deploy)
 
 MavenArtifact.DeployingMainArtifact=Hauptartefakt {0} wird ausgebracht (deploy)
-MavenArtifact.DeployingAttachedArtifact=Verkn�pfte Artefakte {0} werden ausgebracht (deploy)
+MavenArtifact.DeployingAttachedArtifact=Verkn�pfte Artefakte {0} werden ausgebracht (deploy)
```
Comment (Vlatombe) [inline src/main/resources/hudson/maven/reporters/Messages_de.properties:30]: Encoding issue?

File: src/main/resources/hudson/maven/reporters/Messages_de.properties [hunk 7]
```diff
@@ -27,27 +27,27 @@ HistoryWidgetImpl.Displayname=Entwicklungsverlauf
...
 
 MavenArtifactArchiver.DisplayName=Artefakte archivieren
-MavenArtifactArchiver.FailedToInstallToMaster=Artefakt konnte nicht auf dem Master installiert werden
+MavenArtifactArchiver.FailedToInstallToMaster=Artefakt konnte nicht auf dem Controller installiert werden
 
-MavenFingerprinter.DisplayName=Fingerabdr�cke aufzeichnen
+MavenFingerprinter.DisplayName=Fingerabdr�cke aufzeichnen
```
Comment (Vlatombe) [inline src/main/resources/hudson/maven/reporters/Messages_de.properties:35]: Encoding issue?

File: src/test/java/hudson/maven/AbstractMavenTestCase.java [hunk 8]
```diff
@@ -44,9 +44,9 @@ public ComputerLauncher createComputerLauncher(EnvVars env) throws URISyntaxExce
         int sz = this.jenkins.getNodes().size();
         final URL slaveJarURL;
         try {
-            slaveJarURL = this.jenkins.getJnlpJars("slave.jar").getURL();
+            slaveJarURL = this.jenkins.getJnlpJars("agent.jar").getURL();
```
[the remark this answers is not in the document]
Comment (twasyl) [reply src/test/java/hudson/maven/AbstractMavenTestCase.java:47]: Not sure to understand your concern here: IIRC both URLs are valid, for backward compatibility. To ensure, I started a Jenkins instance locally to check the URL is working and it is, as seen on this screenshot.

agent jar_url

Does this address your concern?

[the remark this answers is not in the document]
Comment (olamy) [reply src/test/java/hudson/maven/AbstractMavenTestCase.java:47]: ok sounds good. I was not sure about this backward compat but yes might be for life :)

**Comment by twasyl** [comment]
@olamy You requested code changes but I don't see where. Is it still accurate? I also don't get why the `pct` check doesn't work

**Comment by olamy** [comment]
no idea why this pct github action doesn't work and I don't want neither have time to debug gh actions.
I even removed it from master

**Pull request merged**